main.tf

```diff
@@ -388,6 +422,32 @@ resource "aws_route53_record" "atlantis" {
   }
 }
 
+################################################################################
+# EFS
+################################################################################
+
+resource "aws_efs_file_system" "efs" {
+  count = var.enable_ephemeral_storage ? 0 : 1
+  creation_token = var.name
+}
+
+resource "aws_efs_mount_target" "efs-mt" {
+  # didn't use foreach because terraform doesn't know how many subnets will exist on initial create
```

Comment:
I think `count` will have the same limitation here. `for_each` should be preferred more when order is not important. Could you give it a try and update?

Reply:
Thank you for encouraging me to revisit that resource.  I'm not sure why I got stuck initially, however I was able to find a way to get it done with `for_each` :)